examples/src/indexes/indexing_api/indexing.ts

```diff
@@ -0,0 +1,494 @@
+import { PostgresRecordManager } from "@langchain/community/indexes/postgres";
+import { index } from "langchain/indexes";
+import { PGVectorStore } from "@langchain/community/vectorstores/pgvector";
+import { PoolConfig } from "pg";
+import { OpenAIEmbeddings } from "@langchain/openai";
+import { CharacterTextSplitter } from "langchain/text_splitter";
+import { BaseDocumentLoader } from "langchain/document_loaders/base";
+
+// First, follow set-up instructions at
+// https://js.langchain.com/docs/modules/indexes/vector_stores/integrations/pgvector
+
+const config = {
+  postgresConnectionOptions: {
+    type: "postgres",
+    host: "127.0.0.1",
+    port: 5432,
+    user: "myuser",
+    password: "ChangeMe",
+    database: "api",
+  } as PoolConfig,
+  tableName: "testlangchain",
+  columns: {
+    idColumnName: "id",
+    vectorColumnName: "vector",
+    contentColumnName: "content",
+    metadataColumnName: "metadata",
+  },
+};
+
+const vectorStore = await PGVectorStore.initialize(
+  new OpenAIEmbeddings(),
+  config
+);
+
+// Create a new record manager
+const recordManagerConfig = {
+  postgresConnectionOptions: {
+    type: "postgres",
+    host: "127.0.0.1",
+    port: 5432,
+    user: "myuser",
+    password: "ChangeMe",
+    database: "api",
+  } as PoolConfig,
+  tableName: "upsertion_records",
+};
+const recordManager = new PostgresRecordManager(
+  "test_namespace",
+  recordManagerConfig
+);
+
+// Create the schema if it doesn't exist
+await recordManager.createSchema();
+
+// Index some documents
+const doc1 = {
+  pageContent: "kitty",
+  metadata: { source: "kitty.txt" },
+};
+
+const doc2 = {
+  pageContent: "doggy",
+  metadata: { source: "doggy.txt" },
+};
+
+/**
+ * Hacky helper method to clear content. See the `full` mode section to to understand why it works.
```

Comment:
Can you just explain it in this comment?

I think would be better in smaller chunks/separate examples rather than a giant code block